gridded_etl_tools/utils/extractor.py

```diff
@@ -72,6 +72,38 @@ def request(self, *args, **kwargs) -> bool:
         """
 
 
+class HTTPExtractor(Extractor):
+
+    def __init__(self, dm: dataset_manager.DatasetManager, concurrency_limit: int = 8):
+        """
+        Set the host parameter when initializing an HTTPExtractor object
+        Initializes a session within the dataset manager if it hasn't yet been initialized,
+        so that the `request` method works as intended
+
+        Note that `get_session` is therefore a required method for any DatasetManager using
+        the HTTPExtractor class.
+
+        Parameters
+        ----------
+        host
+            Address to connect to for source data
+        """
+        super().__init__(dm, concurrency_limit=concurrency_limit)
+        if not hasattr(dm, "session"):
+            dm.get_session()
+
+    def request(self, remote_file_path: str, local_file_path: str) -> bool:
+        """
+        Request a file from an HTTP Server and save it to disk
+        Requires an active session within the dataset manager
+        """
+        self.dm.info(f"Downloading {local_file_path}")
+        fil_in_mem = self.dm.session.get(remote_file_path)
```

Comment:
Are we sure the files will fit in memory? This could stream so we don't have to load the while into memory at once. If we're sure the files will fit in RAM easily, then maybe we don't need to do that.

Reply:
Good idea, I've tweaked accordingly